katz/src/main/kotlin/katz/typeclasses/Comonad.kt

```diff
@@ -8,7 +16,51 @@ interface Comonad<F> : Functor<F>, Typeclass {
     fun <A, B> coflatMap(fa: HK<F, A>, f: (HK<F, A>) -> B): HK<F, B>
 
     fun <A> extract(fa: HK<F, A>): A
+}
+
+@RestrictsSuspension
+open class ComonadContinuation<F, A : Any>(val CM: Comonad<F>) : Serializable, Continuation<A> {
+
+    override val context = EmptyCoroutineContext
+
+    override fun resume(value: A) {
+        returnedMonad = value
+    }
+
+    override fun resumeWithException(exception: Throwable) {
+        throw exception
+    }
+
+    internal lateinit var returnedMonad: A
+
+    operator suspend fun <B> HK<F, B>.not(): B = bind { this }
 
+    suspend fun <B> HK<F, B>.bind(): B = bind { this }
+
+    suspend fun <B> bind(m: () -> HK<F, B>): B = suspendCoroutineOrReturn { c ->
```

Comment:
is `bind` a good name for this? bind has the semantics of flatMap, whereas this is 'extract'. what about `cobind`?